pkg/controller/common/helpers.go

```diff
@@ -292,18 +292,27 @@ func ValidateIgnition(ignconfig interface{}) error {
 		if report := validate2.ValidateWithoutSource(reflect.ValueOf(cfg)); report.IsFatal() {
 			return errors.Errorf("invalid ignition V2 config found: %v", report)
 		}
-		return nil
 	case ign3types.Config:
 		if reflect.DeepEqual(ign3types.Config{}, cfg) {
 			return nil
 		}
 		if report := validate3.ValidateWithContext(cfg, nil); report.IsFatal() {
 			return errors.Errorf("invalid ignition V3 config found: %v", report)
 		}
-		return nil
 	default:
 		return errors.Errorf("unrecognized ignition type")
 	}
+
+	typedConfig, err := ConvertConfig(ignconfig)
+	if err != nil {
+		return err
+	}
+	for _, file := range typedConfig.Storage.Files {
+		if file.Contents.Compression != nil {
+			return errors.Errorf("compression %q specified for %q, but compression is not supported", file.Contents.Compression, file.Path)
+		}
+	}
+	return nil
```

Comment:
@yuqi-zhang should I put this here since it validates the ignition or in `ValidateMachineConfig` which calls this?

Reply:
No I think we should put it outside, in validatemachineconfig directly, such that only the controller is calling it.

The ignition validation is technically supposed to check the ignition from a syntax perspective (at least that's how I interpret it)

Reply:
Moved